src/zmap.c

```diff
@@ -826,6 +826,10 @@ int main(int argc, char *argv[])
 	if (!zconf.total_allowed) {
 		log_fatal("zmap", "zero eligible addresses to scan");
 	}
+	if (zconf.list_of_ips_count > 0 && 0xFFFFFFFFU / zconf.list_of_ips_count > 100000) {
+		log_warn("zmap", "list of IPs is small compared to address space, performance will suffer");
+		log_warn("zmap", "consider using an allowlist instead");
```

Comment:
This really feels like it should be a single warning message.

Reply:
It just makes for a very long line